app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/DeviceListFragment.java

```diff
@@ -0,0 +1,357 @@
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.LocalFileTransferActivity.showToast;
+
+/**
+ * Part of the local file sharing module, this fragment is responsible for displaying details of
+ * the user device and the list of available peer devices, and once the user taps on a particular
+ * peer device item, connecting to that device and initiating file transfer.
+ *
+ * File transfer involves two phases:
+ * 1) Handshake with the selected peer device, using {@link PeerGroupHandshakeAsyncTask}
+ * 2) After handshake, starting the files transfer using {@link SenderDeviceAsyncTask} on the sender
+ *    device and {@link ReceiverDeviceAsyncTask} files receiving device
+ *
+ * The starting point for the module is {@link LocalFileTransferActivity}
+ * */
+public class DeviceListFragment extends ListFragment implements WifiP2pManager.PeerListListener, WifiP2pManager.ConnectionInfoListener {
+
+  public static final String TAG = "DeviceListFragment";
+  public static int PEER_HANDSHAKE_PORT = 8009;
+  public static int FILE_TRANSFER_PORT = 8008;
+
+  private SharedPreferenceUtil sharedPreferenceUtil;
+
+  private LocalFileTransferActivity localFileTransferActivity;  // Parent activity, starting point of the module
+  private TransferProgressFragment transferProgressFragment;    // Sibling fragment, for displaying transfer progress
+
+  private View deviceListFragmentRootView = null; // Root view corresponding to the DeviceListFragment
+
+  private boolean fileSender = false; // Whether file sending device or not
+  private ArrayList<Uri> fileUriList; // Uris of files to be shared, available only for the sender device
+  private int totalFilesForTransfer = -1;
+  private int filesSent = 0;          // Count of number of files transferred until now
+  private ArrayList<FileItem> filesToSend = new ArrayList<>();
+
+  private WifiP2pDevice userDevice;   // Represents the device on which the app is running
+  private WifiP2pInfo groupInfo;      // Corresponds to the WiFi P2P group formed between the two devices
+  private List<WifiP2pDevice> peerDevices = new ArrayList<WifiP2pDevice>();
+
+  private WifiP2pDevice selectedPeerDevice = null;
+  private InetAddress selectedPeerDeviceInetAddress;
+
+  private InetAddress fileReceiverDeviceAddress;  // IP address of the file receiving device
+  private boolean fileTransferStarted = false;
+
+  @Override
+  public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+    super.onActivityCreated(savedInstanceState);
+    localFileTransferActivity = (LocalFileTransferActivity) getActivity();
+
+    // As DeviceListFragment extends ListFragment for the purpose of displaying list of peers
+    this.setListAdapter(new WifiPeerListAdapter(localFileTransferActivity, R.layout.row_peer_device, peerDevices));
+
+    if(localFileTransferActivity.isFileSender()) {
```

Comment:
Shouldn't this be ```if (local```

Reply:
I never keep a space between a function-name / if / loop and the bracket - Is this an issue?

Reply:
It is a style issue yes, are you using the [square style](https://github.com/kiwix/kiwix-android/blob/develop/docs/codestyle.md)?

Reply:
I haven't downloaded it, will do so and update the PR
Just choosing the style in android studio will do?

Reply:
You will need to apply the style by reformatting your code. right click on a package and select reformat

Reply:
I personally have a macro that does it when I press `ctrl+s`

Reply:
@Aditya-Sood could you apply the formatter now, it might help with more parts of the review, I imagine single line case statements are not allowed as are single line if statements.

Reply:
@macgills I have [reformatted](https://github.com/kiwix/kiwix-android/pull/1276/commits/57e3aa018264dd8dc773ea2606f57cf2176b38b7) the java files
Thanks for the tip on adding macro, I've added it to `ctrl+shift+s`

Reply:
What should I do about the longer descriptive comments for class fields?
Add them before the declaration,
```
// Uris of files to be shared, available only for the sender device
private ArrayList<Uri> fileUriList;
```
or ignore the code style and leave them at the end of the declaration:
```
private ArrayList<Uri> fileUriList;    // Uris of files to be shared, available only for the sender device
```

Reply:
Well my general comment hating approach is to be as descriptive with the names as possible so the comments are not needed. `urisToBeSharedWithSender`
Don't ignore the codestyle, are you talking about the comment exceeding the maximum width of a line? In that case yes but it above.

Reply:
Cool, I'll correct the comment alignments in the end